omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +257,176 @@ else if(identifierSourceType.equals(Identifier_Pool)){
         }
         return ConversionUtil.convertToRepresentation(savedIdentifierSource, Representation.DEFAULT);         
     }
+    
+    @Override
+    public Object update(String uuid, SimpleObject updateBody, RequestContext context) throws ResponseException {
+        Object updatedIdentifierSource = null;
+        Object batchSizeToUpload = updateBody.get("batchSizeToUpload");
+        Object reservedIdentifiersToUpload = updateBody.get("reservedIdentifiers");
+        Object identifiersToUpload = updateBody.get("identifiersToUpload");
+        Object generateIdentifiers = updateBody.get("generateIdentifiers");
+        SimpleObject output = new SimpleObject();
+        if (reservedIdentifiersToUpload != null) {
+            Object sourceUuid = updateBody.get("sourceUuid");
+            if(sourceUuid != null && Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString()) != null){
+                IdentifierSource source = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                List<String> reservedIdentifiersList = (ArrayList) reservedIdentifiersToUpload;
+                for (int counter = 0; counter < reservedIdentifiersList.size(); counter++) {
+                    if (StringUtils.isNotBlank(reservedIdentifiersList.get(counter))) {
+                        source.addReservedIdentifier(reservedIdentifiersList.get(counter));
+                    }
+                }
+                Context.getService(IdentifierSourceService.class).saveIdentifierSource(source);
+                output.add("message", "Successfully uploaded reserved identifiers.");
+                return output;
+            }
+        }
+        if (batchSizeToUpload != null) {
+            Object sourceUuid = updateBody.get("sourceUuid");
+            if(sourceUuid != null && Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString()) != null){
```

Comment:
Is it necessary to call this twice? Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString())

Reply:
fixed